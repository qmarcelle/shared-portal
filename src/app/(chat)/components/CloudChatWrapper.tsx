'use client';

// CloudChatWrapper injects the Genesys Cloud Messenger SDK and manages Messenger-specific events.
// All script loading, Messenger events, and errors are logged for traceability and debugging.

import { useEffect } from 'react';
import { useChatStore } from '../stores/chatStore';

export default function CloudChatWrapper() {
  const { userData } = useChatStore();

  useEffect(() => {
    // Build chatSettings from env and userData
    window.chatSettings = {
      bootstrapUrl: process.env.NEXT_PUBLIC_GENESYS_BOOTSTRAP_URL || '',
      widgetUrl: process.env.NEXT_PUBLIC_GENESYS_WIDGET_URL || '',
      clickToChatJs: process.env.NEXT_PUBLIC_GENESYS_CLICK_TO_CHAT_JS || '',
      clickToChatEndpoint: process.env.NEXT_PUBLIC_CLICK_TO_CHAT_ENDPOINT || '',
      chatTokenEndpoint: process.env.NEXT_PUBLIC_CHAT_TOKEN_ENDPOINT || '',
      coBrowseEndpoint: process.env.NEXT_PUBLIC_COBROWSE_LICENSE_ENDPOINT || '',
      opsPhone: process.env.NEXT_PUBLIC_OPS_PHONE || '',
      opsPhoneHours: process.env.NEXT_PUBLIC_OPS_HOURS || '',
      ...userData,
    };
    console.debug('[CloudChatWrapper] chatSettings:', window.chatSettings);
  }, [userData]);

  useEffect(() => {
    // Load Genesys Cloud bootstrap and widget scripts
    const bootstrapUrl = window.chatSettings?.bootstrapUrl || '';
    const widgetUrl = window.chatSettings?.widgetUrl || '';
    if (!bootstrapUrl) return;
    const bootstrapScript = document.createElement('script');
    bootstrapScript.src = bootstrapUrl;
    bootstrapScript.async = true;
    bootstrapScript.onload = () => {
      console.debug('[CloudChatWrapper] Genesys bootstrap loaded');
      if (!widgetUrl) return;
      const widgetScript = document.createElement('script');
      widgetScript.src = widgetUrl;
      widgetScript.async = true;
      widgetScript.onload = () => {
        console.debug('[CloudChatWrapper] Genesys widgets loaded');
      };
      widgetScript.onerror = (e) => {
        console.error('[CloudChatWrapper] Failed to load widgets script', e);
      };
      document.body.appendChild(widgetScript);
    };
    bootstrapScript.onerror = (e) => {
      console.error('[CloudChatWrapper] Failed to load bootstrap script', e);
    };
    document.body.appendChild(bootstrapScript);

    // Provide openGenesysChat logic
    window.openGenesysChat = () => {
      if (window.Genesys) {
        try {
          window.Genesys('command', 'Messenger.open');
        } catch (e) {
          console.error(
            '[CloudChatWrapper] Error opening Genesys Messenger',
            e,
          );
        }
      }
    };

    return () => {
      // Cleanup scripts and chatSettings
      document.body.removeChild(bootstrapScript);
      if (widgetUrl) {
        const widgetScript = document.querySelector(
          'script[src="' + widgetUrl + '"]',
        );
        if (widgetScript) document.body.removeChild(widgetScript);
      }
      if (window.chatSettings) delete window.chatSettings;
      if (window.openGenesysChat) delete window.openGenesysChat;
      console.debug('[CloudChatWrapper] Cleanup complete');
    };
  }, [userData]);

  return null;
}
