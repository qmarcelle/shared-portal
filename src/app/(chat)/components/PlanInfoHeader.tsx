'use client';
import { useChatStore } from '@/app/(chat)/stores/chatStore';
import { Title } from '@/components/foundation/Title';

/**
 * Show current plan when multiple plans exist
 */
export default function PlanInfoHeader() {
  const { formInputs } = useChatStore();
  if (formInputs.length <= 1) return null;
  const plan = formInputs.find((f) => f.id === 'PLAN_ID');

  return (
    <Title
      className="plan-info-header"
      text={`You are chatting about plan ${plan?.value || ''}`}
    />
  );
}
