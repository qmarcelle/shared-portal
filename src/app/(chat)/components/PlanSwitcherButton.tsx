'use client';
import { useChatStore } from '@/app/(chat)/stores/chatStore';
import { Button } from '@/components/foundation/Button';
import { usePlanStore } from '../../../userManagement/stores/planStore';

/**
 * "Switch Plan" inside pre-chat window
 */
export default function PlanSwitcherButton() {
  const { endChat } = useChatStore();
  const { plans, selectedPlanId, openPlanSwitcher } = usePlanStore();

  if (plans.length <= 1) return null;

  const handleSwitch = () => {
    endChat();
    openPlanSwitcher();
  };

  return (
    <Button
      label="Switch Plan"
      callback={handleSwitch}
      className="plan-switch-button"
      type="secondary"
    />
  );
}
