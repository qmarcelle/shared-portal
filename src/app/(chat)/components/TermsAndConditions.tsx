'use client';
// import TCs from '@/app/@chat/config/terms'; // { default: '…', lobA: '…', … }
import { useChatStore } from '@/app/(chat)/stores/chatStore';
import { RichText } from '@/components/foundation/RichText';

/**
 * Render T&C based on LOB
 */
export default function TermsAndConditions() {
  const { userData } = useChatStore();
  // const text = TCs[userData.LOB] || TCs.default;
  return (
    <RichText
      spans={['Terms and Conditions']}
      className="terms-and-conditions"
      type="body-2"
    />
  );
}
