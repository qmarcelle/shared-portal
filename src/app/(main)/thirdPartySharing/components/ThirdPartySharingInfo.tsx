import { UpdateRowWithStatus } from '@/components/composite/UpdateRowWithStatus';
import { useAppModalStore } from '@/components/foundation/AppModal';
import { Card } from '@/components/foundation/Card';
import { Column } from '@/components/foundation/Column';
import { Dropdown } from '@/components/foundation/Dropdown';
import { Header } from '@/components/foundation/Header';
import { Row } from '@/components/foundation/Row';
import { Spacer } from '@/components/foundation/Spacer';
import { TextBox } from '@/components/foundation/TextBox';
import Image from 'next/image';
import deleteIcon from '../../../../../public/assets/delete.svg';
import { ThirdPartySharingJourney } from './journeys/ThirdPartySharingJourney';

export const ThirdPartySharingInfo = () => {
  const thirdPartyAppList = [
    {
      appName: 'MyFHR App',
      connectedDate: '01/01/2023',
    },
    {
      appName: 'Erlanger Hospital',
      connectedDate: '01/01/2023',
    },
  ];
  const { showAppModal } = useAppModalStore();
  return (
    <Card className="large-section ">
      <Column>
        <Header className="title-2" text="Linked Apps & Websites" />
        <Spacer size={16} />
        <TextBox text="Below are the apps and websites you’re sharing your information with." />
        <Spacer size={32} />
        <Row>
          <TextBox className="body-1" text="Plan: " />
          <Spacer axis="horizontal" size={8} />
          <Dropdown
            onSelectCallback={() => {}}
            initialSelectedValue="0"
            items={[
              { label: 'Subscriber ID ABC123456789', value: '0' },
              { label: 'Subscriber ID ABC000000000', value: '1' },
            ]}
          />
        </Row>
        <Spacer size={32} />

        {thirdPartyAppList.length > 0 ? (
          thirdPartyAppList.map((item, index) => (
            <Column key={index}>
              <UpdateRowWithStatus
                onClick={() =>
                  showAppModal({
                    content: (
                      <ThirdPartySharingJourney appName={item.appName} />
                    ),
                  })
                }
                label={<TextBox className="font-bold" text={item.appName} />}
                subLabel={'Connected On ' + item.connectedDate}
                methodName="Remove"
                divider={index == thirdPartyAppList.length - 1 ? false : true}
                onOffLabelEnabled={false}
                icon={<Image src={deleteIcon} alt="delete icon"></Image>}
              />
              <Spacer size={32} />
            </Column>
          ))
        ) : (
          <Card className="neutral container p-4 ">
            <Column>
              <TextBox text="You haven’t shared your health information with any third-party apps or websites." />
            </Column>
          </Card>
        )}
      </Column>
    </Card>
  );
};
