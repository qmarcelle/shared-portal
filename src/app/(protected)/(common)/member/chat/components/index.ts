/**
 * Component Exports
 * Centralizes all component exports following the new consolidated structure
 */

// Main Components
export { ChatTrigger } from './ChatTrigger';
export { ChatWidget } from './ChatWidget';
export { PlanInfoHeader } from './PlanInfoHeader';

// Error Handling
export { ChatErrorBoundary } from './shared/ChatErrorBoundary';

// Integration Components (to be consolidated)
export { GenesysScripts } from './GenesysScripts';
