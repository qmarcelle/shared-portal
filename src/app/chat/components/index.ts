// Core Components
export * from './core/ChatButton';
export * from './core/ChatErrorBoundary';
export * from './core/ChatHeader';
export * from './core/ChatWidget';

// Business Hours Components
export * from './business-hours/BusinessHoursNotification';

// Eligibility Components
export * from './eligibility/EligibilityCheck';

// Plan Switcher Components will be added here when implemented
// export * from './plan-switcher/PlanSwitcher';
