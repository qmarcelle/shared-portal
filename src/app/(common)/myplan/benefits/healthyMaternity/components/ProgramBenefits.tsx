/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable react/jsx-key */
import { Column } from '@/components/foundation/Column';
import { Header } from '@/components/foundation/Header';
import { Row } from '@/components/foundation/Row';
import { Spacer } from '@/components/foundation/Spacer';
import { IComponent } from '../../../components/IComponent';
import { TextBox } from '../../../components/foundation/TextBox';
import { BenefitDetails } from '../models/benefit_details';

interface ProgramBenefitsProps extends IComponent {
  benefits: BenefitDetails[];
}

export const ProgramBenefits = ({ benefits }: ProgramBenefitsProps) => {
  return (
    <>
      <Spacer size={12} />
      <Header className="title-2 px-3" text="Program Benefits"></Header>
      <Spacer size={25} />
      <Row className="hidden sm:flex px-3">
        {benefits.slice(0, benefits.length).map((item, index) => (
          <Column className="md:pr-4 md:w-1/3">
            {item.benefitIcon}
            <Spacer size={5} />
            <TextBox text={item.benefitLabel} className="font-bold" />
            <Spacer size={5} />
            <TextBox text={item.benefitCopy} />
          </Column>
        ))}
      </Row>
      <Column className="block sm:hidden px-3">
        {benefits.slice(0, benefits.length).map((item, index) => (
          <Column className="pr-4 mb-4">
            {item.benefitIcon}
            <Spacer size={5} />
            <TextBox text={item.benefitLabel} className="font-bold" />
            <Spacer size={5} />
            <TextBox text={item.benefitCopy} />
          </Column>
        ))}
      </Column>
    </>
  );
};
