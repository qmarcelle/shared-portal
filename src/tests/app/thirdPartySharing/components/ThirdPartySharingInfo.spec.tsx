import { ThirdPartySharingInfo } from '@/app/(protected)/(common)/member/thirdPartySharing/components/ThirdPartySharingInfo';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';

const renderUI = () => {
  return render(<ThirdPartySharingInfo />);
};

describe('Third Party Sharing Info Component', () => {
  it('should render UI correctly', () => {
    const component = renderUI();
    screen.getByRole('heading', { name: 'Linked Apps & Websites' });
    screen.getByText(
      'Below are the apps and websites youâ€™re sharing your information with.',
    );

    expect(component.baseElement).toMatchSnapshot();
  });
});
