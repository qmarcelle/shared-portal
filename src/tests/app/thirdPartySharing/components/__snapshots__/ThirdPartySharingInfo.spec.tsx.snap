// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Third Party Sharing Info Component should render UI correctly 1`] = `
<body>
  <div>
    <div
      class="card-main large-section "
    >
      <div
        class="flex flex-col"
      >
        <h1
          class="title-1 title-2"
        >
          Linked Apps & Websites
        </h1>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
        />
        <p
          class="body-1"
          style="display: block; max-width: 100%;"
        >
          Below are the apps and websites youâ€™re sharing your information with.
        </p>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
        />
        <div
          class="flex flex-row"
        >
          <p
            class="body-1 body-1"
            style="display: block; max-width: 100%;"
          >
            Plan: 
          </p>
          <span
            style="display: block; width: 8px; min-width: 8px; height: 1px; min-height: 1px;"
          />
          <div
            class="relative"
          >
            <div
              class="flex flex-row link "
              tabindex="0"
            >
              <p
                class="body-bold"
              >
                Subscriber ID ABC123456789
              </p>
              <span
                style="display: block; width: 8px; min-width: 8px; height: 1px; min-height: 1px;"
              />
              <img
                alt=""
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                style="color: transparent;"
                width="40"
              />
            </div>
            <section
              class="card-elevated z-20 dropdown-section"
              style="display: none; position: absolute;"
            >
              <div
                class="flex flex-row dropdown-item-selected dropdown-item p-2 cursor-pointer"
              >
                <div
                  class="size-5 mx-2 "
                >
                  <img
                    alt=""
                    class="size-5"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    width="40"
                  />
                </div>
                <div>
                  <p
                    class="whitespace-nowrap mx-2"
                  >
                    Subscriber ID ABC123456789
                  </p>
                </div>
              </div>
              <div
                class="flex flex-row  dropdown-item p-2 cursor-pointer"
              >
                <div
                  class="size-5 mx-2"
                />
                <div>
                  <p
                    class="whitespace-nowrap mx-2"
                  >
                    Subscriber ID ABC000000000
                  </p>
                </div>
              </div>
            </section>
          </div>
        </div>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
        />
        <div
          class="flex flex-col"
        >
          <div
            class="flex flex-col"
          >
            <div
              class="flex flex-row"
            >
              <div
                class="flex flex-col"
              />
              <span
                style="display: block; width: 16px; min-width: 16px; height: 1px; min-height: 1px;"
              />
              <div
                class="flex flex-col"
              >
                <p
                  class="body-1 font-bold"
                  style="display: block; max-width: 100%;"
                >
                  MyFHR App
                </p>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                />
                <p
                  class="body-1 mb-4"
                  style="display: block; max-width: 100%;"
                >
                  Connected On 01/01/2023
                </p>
                <button
                  class="flex flex-row link-container font-bold primary-color pl-0"
                  style="max-width: max-content; height: auto; display: flex;"
                  type="button"
                >
                  <p
                    class="link !no-underline"
                    style="display: flex;"
                  >
                    Remove
                  </p>
                  <p
                    class="ml-1"
                    style="display: flex;"
                  >
                    <img
                      alt=""
                      data-nimg="1"
                      decoding="async"
                      height="40"
                      loading="lazy"
                      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                      style="color: transparent;"
                      width="40"
                    />
                  </p>
                </button>
              </div>
            </div>
            <div
              class="mt-8"
              style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
            />
          </div>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
          />
        </div>
        <div
          class="flex flex-col"
        >
          <div
            class="flex flex-col"
          >
            <div
              class="flex flex-row"
            >
              <div
                class="flex flex-col"
              />
              <span
                style="display: block; width: 16px; min-width: 16px; height: 1px; min-height: 1px;"
              />
              <div
                class="flex flex-col"
              >
                <p
                  class="body-1 font-bold"
                  style="display: block; max-width: 100%;"
                >
                  Erlanger Hospital
                </p>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                />
                <p
                  class="body-1 mb-4"
                  style="display: block; max-width: 100%;"
                >
                  Connected On 01/01/2023
                </p>
                <button
                  class="flex flex-row link-container font-bold primary-color pl-0"
                  style="max-width: max-content; height: auto; display: flex;"
                  type="button"
                >
                  <p
                    class="link !no-underline"
                    style="display: flex;"
                  >
                    Remove
                  </p>
                  <p
                    class="ml-1"
                    style="display: flex;"
                  >
                    <img
                      alt=""
                      data-nimg="1"
                      decoding="async"
                      height="40"
                      loading="lazy"
                      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                      style="color: transparent;"
                      width="40"
                    />
                  </p>
                </button>
              </div>
            </div>
          </div>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
